### PR TITLE
Add ChainAdmin and AccessControlRestriction tests

### PR DESCRIPTION
**Pull request by Yberjon**
# What ❔

This PR adds tests for ChainAdmin and AccessControlRestriction contracts.

### CODE REVIEW

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            abi.encodePacked(
+                "AccessControl: account ",
+                Strings.toHexString(uint160(randomCaller), 20),
+                " is missing role ",
+                Strings.toHexString(uint256(DEFAULT_ADMIN_ROLE), 32)
+            )
+        );
+
+        vm.expectRevert(bytes(revertMsg));
+        vm.startPrank(randomCaller);
```
Comment (saxenism) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol:62]: either use stopPrank with startPrank or just use prank

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/ChainAdmin.t.sol [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        Call[] memory calls = new Call[](2);
+        calls[0] = Call({target: address(gettersFacet), value: 0, data: abi.encodeCall(gettersFacet.getAdmin, ())});
+        calls[1] = Call({target: address(gettersFacet), value: 0, data: abi.encodeCall(gettersFacet.getVerifier, ())});
+
+        vm.startPrank(owner);
+        chainAdmin.multicall(calls, true);
+    }
+
+    function packSemver(uint32 major, uint32 minor, uint32 patch) public returns (uint256) {
+        uint256 SEMVER_MINOR_VERSION_MULTIPLIER = 4294967296;
```
Comment (saxenism) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/ChainAdmin.t.sol:121]: Please do not choose random values, rather use fuzzed values here.

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/ChainAdmin.t.sol [hunk 3]
```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/ChainAdmin.t.sol:3]: it seems to me that this file lacks tests for the `_validateCall` function, i.e. no tests for the case when a restriction forbids certain calls

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+        Call memory call = Call({
+            target: address(chainAdmin),
+            value: 0,
+            data: abi.encodeCall(IChainAdmin.getRestrictions, ())
+        });
+        restriction.validateCall(call, randomCaller);
+    }
+
+    function test_validateCallAccessToFunctionDenied(bytes32 role) public {
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol:147]: what is the difference from test_setRequiredRoleForCallAccessToFunctionDenied?

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        vm.startPrank(owner);
+        restriction.setRequiredRoleForFallback(address(chainAdmin), role);
+        restriction.grantRole(role, randomCaller);
+        vm.stopPrank();
+
+        Call memory call = Call({target: address(chainAdmin), value: 0, data: ""});
+        restriction.validateCall(call, randomCaller);
+    }
+
+    function test_validateCAllAccessToFallbackDenied(bytes32 role) public {
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/AccessControlRestriction.t.sol:180]:
```suggestion
    function test_validateCallAccessToFallbackDenied(bytes32 role) public {
```

**Review by StanislavBreadless — APPROVED** [review]

**Comment by StanislavBreadless** [comment]
I am not sure why CI fails, I'll assume it is a temporary CI issue and will merge

**Pull request merged**